wiki/sensing/speech-recognition.md

```diff
@@ -17,3 +18,120 @@ Here are a few resources to implement speech recognition:
 > Note: Google's Speech API requires a key which can be found in the source code of this package. You can bypass the use of this entire package if you don't wish to have this additional layer between Google's Speech API and your script.
 The source code with examples can be found [here.](https://github.com/Uberi/speech_recognition)
 The key can be found [here.](https://github.com/Uberi/speech_recognition/blob/master/speech_recognition/__init__.py#L613)
+
+## Example
+Here is a simple example of how to use the Speech Recognition package:
+```python
+import speech_recognition as sr
+
+# obtain audio from the microphone
+r = sr.Recognizer()
+
+# Calibrate the microphone
+with sr.Microphone() as source:
+    print("Calibrating microphone...")
+    r.adjust_for_ambient_noise(source, duration=5)
+    print("Calibration complete!")
+
+with sr.Microphone() as source:
+    print("Say something!")
+    audio = r.listen(source)
+
+# recognize speech using Google Speech Recognition
+try:
+    print("Google Speech Recognition thinks you said " + r.recognize_google(audio))
+except sr.UnknownValueError:
+    print("Google Speech Recognition could not understand audio")
+except sr.RequestError as e:
+    print("Could not request results from Google Speech Recognition service; {0}".format(e))
+```
+
+## Hotword/Wakeword Detection
+
+Hotword detection is a very useful feature to have in a robot. It allows the robot to be activated by a simple voice command. This can be used to wake up the robot from sleep mode or to activate a particular function. For example, you can use hotword detection to wake up the robot from sleep mode and then use speech recognition to give it a command.
+
+### Snowboy
+
+Snowboy is a hotword detection engine. It is available [here](https://github.com/seasalt-ai/snowboy). It is available for Python, C++ and Java.
+
+### Porcupine
+
+Porcupine is a hotword detection engine. It is available [here](https://picovoice.ai/platform/porcupine/). Note that this is a paid service with a high level of accuracy.
+
+### Keyword Spotting
+
+Few-shot Keyword Spotting in Any Language and Multilingual Spoken Word Corpus by Harvard University. Follow this [link](https://github.com/harvard-edge/multilingual_kws) to know more.
+
+## Microphone selection
```

Comment:
Casing is inconsistent across headings. Please capitalize each heading through the whole page.

```suggestion
## Microphone Selection
```